scripts/encoding_pathway_analysis.R

```diff
@@ -0,0 +1,92 @@
+# Pan-Cancer Variational Autoencoder
+# Gregory Way 2017
+# scripts/encoding_pathway_analysis.R
+#
+# Performs Overrepresentation Pathway Analyses (ORA)
+#
+# Usage: Run in command line
+#
+#     Rscript --vanilla scripts/gestalt_pathway_analysis.R
+#
+# Output:
+# ORA results for:
+# 1) HGSC-specific feature encodings identified by latent space arithmetic
+# 2) SKCM distinguishing encodings
+
+library(WebGestaltR)
+library(dplyr)
+
+run_webgestalt <- function(genes, output_name, out_dir, background_file) {
+  # Function to run a WebGestalt Pathway Analysis
+  #
+  # Arguments:
+  #    genes: A character of genes to perform ORA test on
+  #    output_name: The name of the sub folder to save results in
+  #    out_dir: The directory to save all pathway analysis results in
+  #    background_file: A filepath pointing to a .txt file with a single column
+  #                     indicating the background to perform the ORA against
+  #
+  # Output:
+  #    Return overrepresented pathways as a dataframe and write full results
+  #    to the specified folder (`gestalt/output_name`)
+
+  webgestalt_output <-
+    WebGestaltR::WebGestaltR(enrichMethod = "ORA",
+                             organism = "hsapiens",
+                             interestGene = genes,
+                             interestGeneType = "genesymbol",
+                             minNum = 4,
+                             fdrMethod = "BH",
+                             is.output = TRUE,
+                             outputDirectory = out_dir,
+                             referenceGeneFile = background_file,
+                             referenceGeneType = "genesymbol",
+                             projectName = output_name)
+  return(webgestalt_output)
+}
+
+# 1) HGSC Analysis
+
+# Specify filenames
+hgsc_out_dir <- file.path("results", "pathway", "hgsc")
+bg_file <- file.path("data", "background_genes.txt")
+hgsc_files <- list.files("results", pattern = "hgsc_node", full.names = TRUE)
+
+# Perform pathway analysis
+for (hgsc_file in hgsc_files) {
+  # Process the base name for the genes being processed
+  hgsc_info <- unlist(strsplit(hgsc_file, "_"))
+  node <- hgsc_info[2]
+  direction <- hgsc_info[3]
+  direction <- unlist(strsplit(direction, "[.]"))[1]
+  base_name <- paste(node, direction, sep = "_")
+  
+  # Read in hgsc genes file
+  hgsc_node_df <- readr::read_tsv(hgsc_file)
+  hgsc_genes <- hgsc_node_df$genes
+
+  # Perform pathway analysis with WebGestaltR
+  output <- run_webgestalt(hgsc_genes, base_name, hgsc_out_dir, bg_file)
```

Comment:
Same question about assignment as below.

Reply:
I will standardize the output variable to remove confusion. Response to this specific point is given below